File: services/order-service/src/main/java/com/binhphuc/order_service/service/CategoryService.java
package com.binhphuc.product_service.service;

import com.binhphuc.product_service.dto.category.CreateCategoryRequest;
import com.binhphuc.product_service.dto.category.CreateCategoryResponse;

public interface CategoryService {
  CreateCategoryResponse create(CreateCategoryRequest createCategoryRequest);
}

File: frontend/src/services/category.service.ts
import { API_ENDPOINTS } from '@/api/endpoints'
import { postJson } from '@/api/request'
import {
  createCategoryRequestSchema,
  createCategoryResponseSchema,
  type CreateCategoryRequest,
  type CreateCategoryResponse,
} from '@/dto/category/create-category'

export const categoryService = {
  create: (
    body: CreateCategoryRequest,
    signal?: AbortSignal,
  ): Promise<CreateCategoryResponse> =>
    postJson({
      url: API_ENDPOINTS.categories.create,
      body,
      requestSchema: createCategoryRequestSchema,
      responseSchema: createCategoryResponseSchema,
      signal,
    }),
}
